### ISSUE TITLE
yarascan Cannot scan for a specific character,Output no result!

### ISSUE THREAD

**Issue by teatreee** [volatilityfoundation/volatility3#372]
**Describe the bug**
I have used volatility 2.6 and volatility 3 for a while.There is a problem when using Yarascan. It can output normally in 2.6, but there is no display in 3. I would like to ask if it is because of my setting problem or other problems?I analyzed the virtual image file in VMEM format.
**Context**
Volatility Version:  volatility3-1.0.0-beta.1
Operating System:  win 10
Python Version:  3.8
Suspected Operating System:  win7
Command:  
python vol.py -vv -f Windows7x64.vmem yarascan.YaraScan --yara-file rule_find2.yar
**To Reproduce**

1. Use command 'python vol.py -vv -f Windows7x64.vmem yarascan.YaraScan --yara-file rule_find2.yar'
2. See error
Offset  Rule

**Expected behavior**

**Screenshots**
![Snipaste_2020-11-08_20-22-27](https://user-images.githubusercontent.com/69066433/98464811-4cdf9e80-2200-11eb-80e0-34aaf9ca1243.png)
![Snipaste_2020-11-08_20-23-07](https://user-images.githubusercontent.com/69066433/98464813-50732580-2200-11eb-9bdc-6885b2e5fb98.png)

**Additional information**
Add any other information about the problem here.

**Comment by ikelos** [comment]
Please could you provide the `rule_find2.yar` file that you're using so we can see what type of query you're trying to make?

**Comment by teatreee** [comment]
Shown in version 2.6
![Snipaste_2020-11-08_20-29-55](https://user-images.githubusercontent.com/69066433/98465168-58cc6000-2202-11eb-98f0-e0862dd2867d.png)
rule_find2.yar
rule keytryc0
{
strings:
        $c0 = {03 00 00 00 00 00 00 30}
condition:
        $c0
}

**Comment by ikelos** [comment]
So, I've been able to get results with the `rule_find2.yar` file you provided.  I also noticed that the debug output from the version you're using is a little old.  Please could you update to the latest version of volatility and confirm that the issue is still present?

**Comment by ikelos** [comment]
There's much less debug output in the latest version now, and in general it's more useful to provide `-vvv` rather than just `-vv`, so if you're still experiencing the issue, please provide that (or `-vvvvvvvv` if you're feeling bold!).  5:)

**Comment by ikelos** [comment]
Here's what the output should look like:
```
Volatility 3 Framework 2.0.0-beta.1
Progress:  100.00		PDB scanning finished                     
Offset	Pid	Rule	Component	Value

0x7fff7d217461	528	keytryc0	$c0	03 00 00 00 00 00 00 30
0x7fff7d217d29	528	keytryc0	$c0	03 00 00 00 00 00 00 30
0x7fff7d217e09	528	keytryc0	$c0	03 00 00 00 00 00 00 30
0x7fff7d217e79	528	keytryc0	$c0	03 00 00 00 00 00 00 30
...
```

**Comment by teatreee** [comment]
Thank you, I have updated to the latest 2.0 version, but there is still no output.And I observed that the output of your VOL3 is different from that of the Vol2.6 version I used. What is the reason for this?
![Snipaste10 30](https://user-images.githubusercontent.com/69066433/98891999-52780580-24da-11eb-8dba-76641659adb7.png)

**Comment by ikelos** [comment]
Volatility 2.6 and Volatility 3 are essentially different programs (complete re-write of the codebase) and in version 3 we're a little more strict about how plugins can return output data, and how the various user interfaces (one of which is the CLI) display that data.  As such, the plugin may output different columns in a different order.

One other consequence of this is that filenames are now passed in a URLs.  We attempt to determine standard files using `urllib.parse.urlparse`.  Unfortunately this detects `C:\Test\rule_find2.yar` as a URL, not a filename (because of the C: at the start), which then causes it to be unable to load the file.  Please try using a relative path, or a complete URL (with `file://` at the start), and we'll look at seeing if we can find a solution to that or at least provide nicer error messages....

**Comment by ikelos** [comment]
Please could you test whether this has been resolved by the same fix as #408 (commit d1b10781) please?

**Comment by ikelos** [comment]
There's been no update on this issue, so I'm going to mark as closed based on #408, but feel free to reopen this if you feel it's still an issue...

**Comment by garanews** [comment]
@ikelos any chance to have in output process name and pid as happened for vol2?
now have only:
```
return renderers.TreeGrid([('Offset', format_hints.Hex), ('Rule', str), ('Component', str), ('Value', bytes)],
                                  self._generator())
                                  ```
